server/src/actors/project.rs

```diff
@@ -412,35 +411,125 @@ pub struct PublicKeyConfig {
 }
 
 /// Data for applying rate limits in Redis
+#[derive(Debug, Clone)]
+pub enum Quota {
+    RejectAll(RejectAllQuota),
+    Redis(RedisQuota),
+}
+
+mod __quota_serialization {
+    use super::{Quota, RedisQuota, RejectAllQuota};
+
+    use serde::{de::Error, Deserialize, Deserializer, Serialize, Serializer};
+
+    #[derive(Serialize, Deserialize, Default)]
+    #[serde(default)]
+    struct QuotaSerdeHelper {
+        #[serde(skip_serializing_if = "Option::is_none")]
+        limit: Option<u64>,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        reason_code: Option<String>,
```

Comment:
_nit_: You can use `Cow<'a, str>` here to avoid clones (it cannot be `&'a str` because then you couldn't deserialize from a Reader)